hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupReaderSchemaHandler.java

```diff
@@ -74,6 +74,10 @@ public class FileGroupReaderSchemaHandler<T> {
   // requiredSchema: the requestedSchema with any additional columns required for merging etc
   protected final Schema requiredSchema;
 
+  // the schema for updates, usually it equals with the requiredSchema,
+  // the only exception is for incoming records, which do not include the metadata fields.
+  protected Schema schemaForUpdates;
```

Comment:
We should think about if we want this in the schema handler. Currently all the fields in the schema handler are final. Would it be better to just move this to the reader context?

Reply:
the intention is to keep all different schemas in one place and thats FG reader SchemaHandler.

Reply:
your intention